js/data.js

```diff
@@ -1,127 +1,116 @@
-  'parking',
-  'washer',
-  'elevator',
-  'conditioner',
-];
+const setHousingPrice = (cb) => {
+  housingPrice.addEventListener('change', () => {
+    cb();
+  });
+};
 
-const PHOTOS_LIST = [
-  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
-  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
-  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg',
-];
+const setHousingRooms = (cb) => {
+  housingRooms.addEventListener('change', () => {
+    cb();
+  });
+};
 
-const createOffer = () => {
-  const getAvatarValue = () => {
-    const value = getRandomInt(1, ARRAY_LENGTH);
-    return (value < 10) ? `0${value}` : `${value}`;
-  };
+const setHousingGuests = (cb) => {
+  housingGuests.addEventListener('change', () => {
+    cb();
+  });
+};
 
-  const getShuffledItems = (items) => {
-    const itemsCopy = items.slice().sort(() => 0.5 - Math.random());
-    return itemsCopy;
-  };
+const setHousingFeatures = (cb) => {
+  for (const housingFeature of housingFeatures) {
+    housingFeature.addEventListener('change', () => {
+      cb();
+    });
+  }
+};
 
-  const getRandomItems = (items) => {
-    const randomIndex = getRandomInt(0, items.length);
-    return getShuffledItems(items).slice(0, randomIndex);
-  };
+const getOfferRank = (obj) => {
+  let rank = 0;
+  for (const housingFeature of housingFeatures) {
+    if (obj.offer.features && housingFeature.checked) {
+      obj.offer.features.forEach((feature) => {
+        if (feature === housingFeature.value) {
+          rank += 1;
+        }
+      });
+    }
+  }
+  return rank;
+};
 
-  const locationLat = getRandomFloat(35.65000, 35.70000, 5);
-  const locationLng = getRandomFloat(139.70000, 139.80000, 5);
+const compareOffers = (offerA, offerB) => {
+  const rankA = getOfferRank(offerA);
+  const rankB = getOfferRank(offerB);
+  return rankB - rankA;
+};
 
-  return {
-    author: {
-      avatar: `img/avatars/user${getAvatarValue()}.png`,
-    },
-    offer: {
-      title: 'Sample text',
-      address: `${locationLat}, ${locationLng}`,
-      price: getRandomInt(100, 1000000),
-      type: OFFER_TYPE[getRandomInt(0, OFFER_TYPE.length - 1)],
-      rooms: getRandomInt(1, 10),
-      guests: getRandomInt(1, 20),
-      checkin: CHECKIN_TIME[getRandomInt(0, CHECKIN_TIME.length - 1)],
-      checkout: CHECKOUT_TIME[getRandomInt(0, CHECKOUT_TIME.length - 1)],
-      features: getRandomItems(FEATURES_LIST),
-      description: 'description',
-      photos: getRandomItems(PHOTOS_LIST),
-    },
-    location: {
-      lat: locationLat,
-      lng: locationLng,
-    },
-  };
+const renderOffers = (offers, mapElement) => {
+  mapElement.clearLayers();
+  offers
+    .filter((obj) => {
+      if (obj.offer.type === housingType.value || housingType.value === 'any') {
+        return true;
+      }
+      return false;
```

Comment:
```suggestion
      return obj.offer.type === housingType.value || housingType.value === 'any'
```
во всех можно упростить до такого